src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -241,13 +241,12 @@ void ObjectMonitor::operator delete[] (void *p) {
   operator delete(p);
 }
 
+#ifdef ASSERT
```

Comment:
There would be less `#ifdef ASSERT` clutter if just the body of check_object_context were conditionalized.  Then the calls wouldn't need to be.  Your call...

Reply:
I'll make that change.